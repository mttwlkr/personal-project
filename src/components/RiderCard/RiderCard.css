.rider-container-card {
  width: 250px;
  height: 500px;
  background: #fff;
  margin: 2em 1em 0em 1em;
  color: #2c3e50;
}

.rider-card-thumbnail {
  width: 100%;
}

h3,
p {
  margin: .8em 0;
}

.rider-card-info {
  text-align: left;
  padding: 1em 1.5em 0em;
}

.rider-card-stat-title,
.rider-card-sponsor-title {
  font-weight: 800;
}

/*.rider-card-sponsor {
  text-align: center;
}*/

.rider-card-stat,
.rider-card-games-played-title {
  padding: .1em 0;
}

.rider-card-games-played-value {
  font-weight: 400;
}

h4 {
  margin: 0;
  padding: 0;
}