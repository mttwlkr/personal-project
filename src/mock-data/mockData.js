export const mockAPIRidersData = [
  {
    "id": 30,
    "name": "Nicky Keefer",
    "sponsor": "Atomic",
    "avatar": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/030/original/keefer.jpg?1419916201",
    "country": "us",
    "birthdate": "1991-02-16",
    "slvsh_avg": 167,
    "url": "http://www.slvsh.com/riders/30.json",
    "age": 27
  },
  {
    "id": 23,
    "name": "Torin Yater-Wallace",
    "sponsor": "Red Bull",
    "avatar": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/023/original/torin.jpg?1417500198",
    "country": "us",
    "birthdate": "1995-12-01",
    "slvsh_avg": 444,
    "url": "http://www.slvsh.com/riders/23.json",
    "age": 22
  },
  {
    "id": 19,
    "name": "Lupe Hagearty",
    "sponsor": "Nordica",
    "avatar": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/019/original/lupe.jpg?1416289155",
    "country": "us",
    "birthdate": "1992-08-13",
    "slvsh_avg": 444,
    "url": "http://www.slvsh.com/riders/19.json",
    "age": 25
  }
]

export const mockAPIGamesData = {
"current_page": 1,
"total_pages": 42,
  "posts": [
    {
      "id": 869,
      "created_at": "2018-04-05T15:16:50.775Z",
      "updated_at": "2018-04-05T15:16:50.775Z",
      "video_host": "youtube",
      "vimeo_id": "dJQmXDZW1ig",
      "type": "Film",
      "location": "Grandvalira, Andorra",
      "video_url": "https://www.youtube.com/embed/dJQmXDZW1ig",
      "title": "Recap Edit || Slvsh Cup&nbsp;Grandvalira",
      "poster_img": "http://slvsh_prod.s3.amazonaws.com/films/poster_imgs/000/000/869/medium/RecapCover16x9.jpg?1522941405",
      "post_class": "film",
      "publish_date": "1 day",
      "href": "/films/869",
      "description": "Thats it folks!! As the 16th and Final game of the cup has made its way online we have to close out this years Slvsh Cup. We can't thank you, the fans, enough for all the views and comments on this years cup. This has been by far our best event to date and we can only hope to do it bigger and better next time! As a bonus and thank you, we give you the #SlvshCupGrandvalira Recap edit. This has to be one of the funnest weeks for all the athletes no matter how they end up in the contest, and this video should prove that! A huge shout out to Grandvalira, Monster Energy and Volkl Skis for their continued support of Slvsh and making this event happen!! Film/edit by Andrew Napier Music PhantomOnDaTrack",
      "formatted_publish_date": "04.05.2018",
      "tags": [
        { "name": "Joss Christensen" },
        { "name": "Alex Bellemare" },
        { "name": "Steve Stepp" },
        { "name": "Johan Berg" },
        { "name": "charlie lasser" },
        { "name": "slvsh cup" },
        { "name": "alex hall" },
        { "name": "jake carney" },
        { "name": "Andrew Napier" },
        { "name": "emil granbom" },
        { "name": "hugo burvall" },
        { "name": "edit" },
        { "name": "slvsh cup grandvalira" },
        { "name": "Quinn Wolferman" },
        { "name": "henrik harlaut" },
        { "name": "Ferdinand Dahl" },
        { "name": "Simone Canal" },
        { "name": "Hunter Hess" },
        { "name": "Ethan Swadburg" },
        { "name": "Tom Ritsch" },
        { "name": "James Woods" },
        { "name": "elias syrja" },
        { "name": "Aleksi Patja" }
      ],
      "tags_string": "Joss Christensen, Alex Bellemare, Steve Stepp, Johan Berg, charlie lasser, slvsh cup, alex hall, jake carney, Andrew Napier, emil granbom, hugo burvall, edit, slvsh cup grandvalira, Quinn Wolferman, henrik harlaut, Ferdinand Dahl, Simone Canal, Hunter Hess, Ethan Swadburg, Tom Ritsch, James Woods, elias syrja, Aleksi Patja"
    },
    {
      "id": 836,
      "created_at": "2018-04-03T13:05:25.506Z",
      "updated_at": "2018-04-03T13:05:25.506Z",
      "video_host": "youtube",
      "vimeo_id": "b_9xjpVljtM",
      "type": "Game",
      "location": "Grandvalira, AD",
      "video_url": "https://www.youtube.com/embed/b_9xjpVljtM",
      "title": "Final || Henrik Harlaut vs. Quinn Wolferman || Slvsh Cup&nbsp;Grandvalira",
      "poster_img": "http://slvsh_prod.s3.amazonaws.com/games/poster_imgs/000/000/836/medium/HenrikvsQuinnCover16x9.jpg?1522760710",
      "post_class": "slvsh",
      "publish_date": "3 days",
      "href": "/games/836",
      "winner": {
        "id": 150,
        "name": "Henrik Harlaut",
        "created_at": "2017-10-10T12:48:54.345Z",
        "updated_at": "2018-03-15T13:23:05.198Z",
        "sponsor": "Monster Energy",
        "avatar_file_name": "henrik600.jpg",
        "avatar_content_type": "image/jpeg",
        "avatar_file_size": 203220,
        "avatar_updated_at": "2017-10-10T12:48:53.810Z",
        "country": "se",
        "birthdate": "1995-01-01",
        "slvsh_avg": 500
      },
      "loser": {
        "id": 110,
        "name": "Quinn Wolferman",
        "created_at": "2016-08-02T07:30:32.665Z",
        "updated_at": "2016-08-02T07:32:11.888Z",
        "sponsor": "Tall Tee",
        "avatar_file_name": "QuinnWolferman.jpg",
        "avatar_content_type": "image/jpeg",
        "avatar_file_size": 51763,
        "avatar_updated_at": "2016-08-02T07:30:32.264Z",
        "country": "us",
        "birthdate": "1992-10-06",
        "slvsh_avg": 500
      },
      "game_type": "S.L.V.S.H.",
      "winner_id": 150,
      "loser_id": 110,
      "winner_avatar=": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/150/thumb/henrik600.jpg?1507639733",
      "loser_avatar=": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/110/thumb/QuinnWolferman.jpg?1470123032",
      "sorted_riders": [
        { 
          "name": "Henrik Harlaut",
          "avatar_url": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/150/thumb/henrik600.jpg?1507639733"
        },
        {
        "name": "Quinn Wolferman",
        "avatar_url": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/110/thumb/QuinnWolferman.jpg?1470123032"
        }
      ],
      "winner_score": 5,
      "loser_score": 5,
      "description": "Grandvalira, Monster Energy, and Volkl Skis are proud to present Slvsh Cup Grandvalira 2018! The final match up and the battle for $10,000 comes down to Henrik Harlaut and Quinn Wolferman. Could we ask for a better match up? Don't think so!",
      "formatted_publish_date": "04.03.2018",
      "tags": [
        { "name": "slvsh cup" },
        { "name": "game" },
        { "name": "Quinn Wolferman" },
        { "name": "henrik harlaut" }
      ],
      "tags_string": "slvsh cup, game, Quinn Wolferman, henrik harlaut"
    },
    {
      "id": 835,
      "created_at": "2018-04-02T14:10:08.000Z",
      "updated_at": "2018-04-02T14:10:08.000Z",
      "video_host": "youtube",
      "vimeo_id": "_ygdXqvSstg",
      "type": "Game",
      "location": "Grandvalira, AD",
      "video_url": "https://www.youtube.com/embed/_ygdXqvSstg",
      "title": "Consolation || Alex Hall vs. Hunter Hess || Slvsh Cup&nbsp;Grandvalira",
      "poster_img": "http://slvsh_prod.s3.amazonaws.com/games/poster_imgs/000/000/835/medium/AlexVHunterCover16x9.jpg?1522678205",
      "post_class": "slvsh",
      "publish_date": "4 days",
      "href": "/games/835",
      "winner": {
        "id": 80,
        "name": "Alex Hall",
        "created_at": "2015-12-08T14:30:14.125Z",
        "updated_at": "2016-08-09T08:52:12.365Z",
        "sponsor": "Shred",
        "avatar_file_name": "Alexhall.jpg",
        "avatar_content_type": "image/jpeg",
        "avatar_file_size": 89077,
        "avatar_updated_at": "2015-12-08T14:30:12.918Z",
        "country": "us",
        "birthdate": "1998-10-21",
        "slvsh_avg": 500
      },
      "loser": {
        "id": 153,
        "name": "Hunter Hess",
        "created_at": "2017-11-14T13:14:45.116Z",
        "updated_at": "2017-11-14T13:20:26.628Z",
        "sponsor": "Volkl",
        "avatar_file_name": "482179_614260521934935_889184546_n.jpg",
        "avatar_content_type": "image/jpeg",
        "avatar_file_size": 266086,
        "avatar_updated_at": "2017-11-14T13:14:44.512Z",
        "country": "us",
        "birthdate": "1997-10-01",
        "slvsh_avg": 500
      },
      "game_type": "S.L.V.S.H.",
      "winner_id": 80,
      "loser_id": 153,
      "winner_avatar=": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/080/thumb/Alexhall.jpg?1449585012",
      "loser_avatar=": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/153/thumb/482179_614260521934935_889184546_n.jpg?1510665284",
      "sorted_riders": [
        {
          "name": "Alex Hall",
          "avatar_url": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/080/thumb/Alexhall.jpg?1449585012"
        },
        {
          "name": "Hunter Hess",
          "avatar_url": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/153/thumb/482179_614260521934935_889184546_n.jpg?1510665284"
        }
      ],
      "winner_score": 5,
      "loser_score": 5,
      "description": "Grandvalira, Monster Energy, and Volkl Skis are proud to present Slvsh Cup Grandvalira 2018! Alex Hall and Hunter Hess battle it out for the 3rd place title in the consolation round. Watch to see who walks away with 3rd place and $2,500.00",
      "formatted_publish_date": "04.02.2018",
      "tags": [
        { "name": "slvsh cup" },
        { "name": "alex hall" },
        { "name": "monster energy" },
        { "name": "sunset park" },
        { "name": "game" },
        { "name": "grandvalira" },
        { "name": "Hunter Hess" }
      ],
      "tags_string": "slvsh cup, alex hall, monster energy, sunset park, game, grandvalira, Hunter Hess"
    }
  ]
} 