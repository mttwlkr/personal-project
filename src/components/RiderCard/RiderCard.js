import React from 'react';
import './RiderCard.css';
import FlagIcon from '../FlagIconFactory/FlagIconFactory';

const RiderCard = ({rider, handleRider}) => {
  const statKeys = Object.keys(rider).filter( key => {
    return key === "Offense" || key === "Defense" || key === "Difficulty" || key === "Overall"
  })
  const displayStats = statKeys.map((stat, index) => {
    return <div className='rider-card-stat' key={index}>
        <span className='rider-card-stat-title'>{`${stat}`}</span>
        {`: ${(rider[stat]).toLocaleString('en-IN', { style: 'percent', maximumSignificantDigits: 3 })}`}
      </div>
  })

  return (
    <div 
      className='rider-container-card'
      onClick={() => handleRider(rider.id)}
    >
      <img 
        className='rider-card-thumbnail' 
        src={rider.avatar}
        alt='slvsh-rider-thumbnail'
      />

      <h3>{rider.name}</h3>
      <FlagIcon 
        code={rider.country} 
        size={'lg'} 
      />

      <section className='rider-card-info'>
        <h4 className='rider-card-games-played-title'>{`SLVSH Games:`}
          <span className='rider-card-games-played-value'>{` ${rider.games_played || 0}`}</span>
        </h4>
        {displayStats}
        <p className='rider-card-sponsor'>
          <span className='rider-card-sponsor-title'>{`Sponsor: `}</span>{`${rider.sponsor}`}
        </p>
      </section>
    </div>
  )
}

export default RiderCard;

// const getStats = () => {

//     const topOffense = () => props.riders.filter(rider => rider.games_played > 1).sort((a, b) => b.Offense - a.Offense).slice(0, 10) 
//     const topDefense = () => props.riders.filter(rider => rider.games_played > 1).sort((a, b) => b.Defense - a.Defense).slice(0, 10)
//     const topDifficulty = () => props.riders.filter(rider => rider.games_played > 1).sort((a, b) => b.Difficulty - a.Difficulty).slice(0, 10)
//     const topOverall = () => props.riders.filter(rider => rider.games_played > 1).sort((a, b) => b.Overall - a.Overall).slice(0, 10)
    
//     const stats = props.riders.reduce((statsObj, rider) => {
//       if (rider.games_played > 1) {
//         statsObj.topOffense = topOffense()
//         statsObj.topDefense = topDefense()
//         statsObj.topDifficulty = topDifficulty()
//         statsObj.topOverall = topOverall()
//       }
//       return statsObj
//     }, {})
//     return stats
//   }
