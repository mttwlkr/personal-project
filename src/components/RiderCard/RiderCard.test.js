import React from 'react';
import RiderCard from './RiderCard.js';
import { mockRiderReducerExpected, mockRiderWithZeroGames } from '../../mock-data/mockData';
import { shallow } from 'enzyme';

describe('RiderCard', () => {

  let wrapper;
  const mockHandleRider = jest.fn();

  it('should match the snapshot', () => {
    wrapper = shallow(<RiderCard 
      rider={mockRiderReducerExpected}
      handleRider={mockHandleRider}
    />);
  });

  it('should not render stats if rider has none', () => {
    wrapper = shallow(<RiderCard 
      rider={mockRiderWithZeroGames}
      handleRider={mockHandleRider}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke handleRider on click', () => {
    wrapper = shallow(<RiderCard 
      rider={mockRiderReducerExpected}
      handleRider={mockHandleRider}
    />);
    wrapper.find('.rider-container-card').simulate('click');
    expect(mockHandleRider).toHaveBeenCalled();
  });
});