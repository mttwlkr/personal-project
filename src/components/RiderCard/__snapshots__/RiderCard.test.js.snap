// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiderCard should not render stats if rider has none 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RiderCard
    handleRider={[Function]}
    rider={
        Object {
            "age": 26,
            "avatar": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/005/original/Noah_Wallace.jpg?1427290713",
            "birthdate": "1991-07-02",
            "country": "us",
            "id": 5,
            "name": "Noah Wallace",
            "slvsh_avg": 489,
            "sponsor": "Giro",
            "url": "http://www.slvsh.com/riders/5.json",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="slvsh-rider-thumbnail"
          className="rider-card-thumbnail"
          src="http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/005/original/Noah_Wallace.jpg?1427290713"
/>,
        <h3>
          Noah Wallace
</h3>,
        <FlagIcon
          code="us"
          size="lg"
/>,
        <section
          className="rider-card-info"
>
          <h4
                    className="rider-card-games-played-title"
          >
                    SLVSH Games:
                    <span
                              className="rider-card-games-played-value"
                    >
                               0
                    </span>
          </h4>
          <p
                    className="rider-card-sponsor"
          >
                    <span
                              className="rider-card-sponsor-title"
                    >
                              Sponsor: 
                    </span>
                    Giro
          </p>
</section>,
      ],
      "className": "rider-container-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "slvsh-rider-thumbnail",
          "className": "rider-card-thumbnail",
          "src": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/005/original/Noah_Wallace.jpg?1427290713",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Noah Wallace",
        },
        "ref": null,
        "rendered": "Noah Wallace",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "code": "us",
          "size": "lg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="rider-card-games-played-title"
>
              SLVSH Games:
              <span
                            className="rider-card-games-played-value"
              >
                             0
              </span>
</h4>,
            Array [],
            <p
              className="rider-card-sponsor"
>
              <span
                            className="rider-card-sponsor-title"
              >
                            Sponsor: 
              </span>
              Giro
</p>,
          ],
          "className": "rider-card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "SLVSH Games:",
                <span
                  className="rider-card-games-played-value"
>
                   0
</span>,
              ],
              "className": "rider-card-games-played-title",
            },
            "ref": null,
            "rendered": Array [
              "SLVSH Games:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " 0",
                  "className": "rider-card-games-played-value",
                },
                "ref": null,
                "rendered": " 0",
                "type": "span",
              },
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="rider-card-sponsor-title"
>
                  Sponsor: 
</span>,
                "Giro",
              ],
              "className": "rider-card-sponsor",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sponsor: ",
                  "className": "rider-card-sponsor-title",
                },
                "ref": null,
                "rendered": "Sponsor: ",
                "type": "span",
              },
              "Giro",
            ],
            "type": "p",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="slvsh-rider-thumbnail"
            className="rider-card-thumbnail"
            src="http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/005/original/Noah_Wallace.jpg?1427290713"
/>,
          <h3>
            Noah Wallace
</h3>,
          <FlagIcon
            code="us"
            size="lg"
/>,
          <section
            className="rider-card-info"
>
            <h4
                        className="rider-card-games-played-title"
            >
                        SLVSH Games:
                        <span
                                    className="rider-card-games-played-value"
                        >
                                     0
                        </span>
            </h4>
            <p
                        className="rider-card-sponsor"
            >
                        <span
                                    className="rider-card-sponsor-title"
                        >
                                    Sponsor: 
                        </span>
                        Giro
            </p>
</section>,
        ],
        "className": "rider-container-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "slvsh-rider-thumbnail",
            "className": "rider-card-thumbnail",
            "src": "http://slvsh_prod.s3.amazonaws.com/riders/avatars/000/000/005/original/Noah_Wallace.jpg?1427290713",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Noah Wallace",
          },
          "ref": null,
          "rendered": "Noah Wallace",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "code": "us",
            "size": "lg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="rider-card-games-played-title"
>
                SLVSH Games:
                <span
                                className="rider-card-games-played-value"
                >
                                 0
                </span>
</h4>,
              Array [],
              <p
                className="rider-card-sponsor"
>
                <span
                                className="rider-card-sponsor-title"
                >
                                Sponsor: 
                </span>
                Giro
</p>,
            ],
            "className": "rider-card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "SLVSH Games:",
                  <span
                    className="rider-card-games-played-value"
>
                     0
</span>,
                ],
                "className": "rider-card-games-played-title",
              },
              "ref": null,
              "rendered": Array [
                "SLVSH Games:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " 0",
                    "className": "rider-card-games-played-value",
                  },
                  "ref": null,
                  "rendered": " 0",
                  "type": "span",
                },
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="rider-card-sponsor-title"
>
                    Sponsor: 
</span>,
                  "Giro",
                ],
                "className": "rider-card-sponsor",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sponsor: ",
                    "className": "rider-card-sponsor-title",
                  },
                  "ref": null,
                  "rendered": "Sponsor: ",
                  "type": "span",
                },
                "Giro",
              ],
              "type": "p",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
